src/rust/engine/Cargo.toml

```diff
@@ -21,6 +21,88 @@ debug = 1
 # Optimise for the speed of our binary, rather than the speed of compilation.
 codegen-units = 1
 
+[lib]
```

Comment:
The diff algorithm produces a sucky result here. The actual changes are appending a `[workspace.dependencies]` table at the end of the file, and replacing various `[dependencies]` versions with `workpace = true`, to refer to the versions in the workspace.

Reply:
I played around with other diff algorithms supported by git diff, but couldn't get anything better.